coral-dbt/build.gradle

```diff
@@ -12,43 +12,74 @@ spotless {
 }
 
 task runCoralDbtTests {
+  def runTest = false
+
+  // Ensure pip is installed
+  try {
+    println("Ensuring pip is installed...")
+    def ensurePipProcessBuilder = new ProcessBuilder([
+      'python3',
+      '-m',
+      'ensurepip',
+      '--upgrade'
+    ])
+    def ensurePipProcess = ensurePipProcessBuilder.start()
+    ensurePipProcess.waitFor()
+  } catch (Exception e) {
+    println("Pip installation failed. Trying dependency installation regardless.")
+  }
+
   // Install Python dependencies
-  def installModulesProcessBuilder = new ProcessBuilder([
-    'python3',
-    '-m',
-    'pip',
-    'install',
-    '-r',
-    'requirements.txt'
-  ])
-  installModulesProcessBuilder.directory(file('src/main/resources/tests'))
-  def installModulesProcess = installModulesProcessBuilder.start()
-  installModulesProcess.waitFor()
+  try {
```

Comment:
this block shouldnt be triggered if the previous try-catch failed?

Reply:
I was thinking in case it is a permissions problem but the user already has pip installed.

Reply:
but we didnt validate if pip is installed.

Reply:
Yes, if that is the case, the second try-catch block will catch that.